### PR TITLE
undo delete of expenses and categories

### PR DESCRIPTION
**Pull request by SeR4h**
@dennisja Please review this PR.Thanks☺

### CODE REVIEW

File: routes/undoDelete.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+const pool = require('../models/db');
+const verifyToken = require('../verifyToken');
+
+const undoDelete = (app) => {
+    // Undo delete of an expense
+    app.delete('/undo-expense/:expenseId', verifyToken.verifyToken, (request, response) => {
```
Comment (dennisja) [inline routes/undoDelete.js:6]: You can use `patch` for this I guess, It makes more sense than a delete

File: routes/undoDelete.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            } else if (result.affectedRows > 0) {
+                pool.query('UPDATE expenses SET deleted="NO"  WHERE created_by = ? AND category_id = ?', [username, categoryId], (err, res) => {
+                    if (res.affectedRows > 0) {
+                        response.send({
+                            code: 200,
+                            message: 'Category and its expenses recovered successfully.',
+                        });
+                    } else {
+                        response.send({
+                            code: 204,
```
Comment (dennisja) [inline routes/undoDelete.js:47]: Why did you use a 204 here?

Comment (SeR4h) [reply routes/undoDelete.js:47]: To emphasize that the category has no expenses attached to it

**Comment by SeR4h** [comment]
@dennisja Please review this.Also this is my final story,what should I do next?host the database or??

**Comment by dennisja** [comment]
You can proceed to host the API and start working on the frontend.
But I have a few questions:
1. Would you be willing to write tests for this code If I gave you an example of how to go about it?
2. Would you be willing to make any architectural changes? We can later rewrite this using the CLEAN architecture.
3. Would you be willing to add some validation to some of the endpoints?

**Comment by SeR4h** [comment]
@dennisja Am willing to learn all..Waiting on the examples for writing tests

**Comment by dennisja** [comment]
Merge this, I'll make you a document to start working on the frontend, If I get time over the weekend I'll write you an example to write tests

**Pull request merged**